README.md
# Basic-Quiz-System

main.c // The main function.

functions.c // Contains functions used to load questions and take user answers for each subject.

count_questions.h // Used to count the number of questions in a file.

grade.h // Contain function used to calculate grade for given percentage.

------------------------------------------------------------------------------
Questions are stored in:-

chemistry.txt

maths.txt

physics.txt

UPDATE: Added random question numbers.

UPDATE 2: Results are now saved in result.txt


history.txt
Who was the first President of the United States of America?
John F.Kennedy
Jimmy Carter 
Thomas Jefferson 
George Washington
D
		
When did Christopher Columbus set sail for the West?
1526
1492
1776
1691
B

When did India come directly under control of British Empire?
1689
1757
1879
1857
D

Who succeeded Akbar as the Emperor of India?
Bahadur Shah
Aurangzeb
Jahangir
Shah Alam
C

Whom did Muhammad bin Qasim defeat in 712 to establish the first Muslim state in subcontinent?
Raja Prithviraj
Raja Dahir
Shivaji
Raja Vijay
B

Name the battle that was responsible for America's entry in World War 2
Pearl Harbor
Midway
Coral Sea
Okinawa
A

Where did the first modern civilizations originate?
Brazil
China
Yemen
Iraq
D

Who was the Muslim leader who repelled the European Crusades?
Mahmud Ghaznavi
Saladin
Amir Timur
Babar
B

When did the Communist Party take power in China?
1951
1947
1962
1949
D

When did the Bolsheviks form the Soviet Union?
1917
1912
1923
1937
A

main.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>
#include <windows.h>

#include "count_questions.h" // Includes fucntion used to count the number of questions in each file.
#include "functions.h" // Include fucntion which displays question and take input from user.
#include "grade.h" // Includes the grade function used to calculate grade by giving percentage as input.

int main ()
{
srand(time(NULL)); // Set the seed for rand function.
char student_name[40];
char full_name[40];
char *first_name;
char roll_no[20];

/*This is going to be our welcome screen.
We as the user to enter his name and roll no so we can print and save the reuslt at end*/

printf ("\t\t\t\tInterrogo Quiz System\n\n");
printf ("\t\t\t Enter your name: ");
fgets (student_name, 40, stdin);
strcpy(full_name, student_name);
first_name = strtok (student_name," "); // Strtok function is used to extract the first name of user.
printf ("\t\t\t Enter your Roll No: ");
scanf ("%s", roll_no);
system ("cls");

/*The maths_quiz and physics_quiz etc variables are used as flag to check
if the quiz for that subject is already taken so that the user can take the
quiz for a subject only one time. Once all subjects are done the loop is exited
and result is printed.
*/
int maths_quiz = 0, maths_correct = 0;
int physics_quiz = 0, physics_correct = 0;
int chemistry_quiz = 0, chemistry_correct = 0;
int history_quiz =0, history_correct = 0;
int biology_quiz = 0, biology_correct = 0;

printf ("Welcome %s!\n\n", first_name);
while (maths_quiz==0 || physics_quiz==0 || chemistry_quiz==0 || history_quiz == 0 || biology_quiz == 0)
{
printf ("Select the subject you want to take quiz\n");
printf ("1. Maths\t\t");
printf ("2. Physics\n");
printf ("3. Chemistry\t\t");
printf ("4. History\n");
printf ("5. Biology\n\n");
printf ("Enter 0 to end the quiz.\n");

int selection;
scanf ("%d", &selection); // Choose the subject.
fflush(stdin);
if (selection==0)
{
    break;   // The while loop is exited when user enter 0.
}


switch (selection)
{
    case 1: if (maths_quiz == 0)
            {   system ("cls");
                maths_correct = maths();
                maths_quiz = 1;
            }
            else
            {   printf ("You have already taken Math's quiz. Choose another option.\n");
                Sleep(1000);
            }
            break;

    case 2: if (physics_quiz == 0)
            {   system ("cls");
                physics_correct = physics();
                physics_quiz = 1;
            }
            else
            {   printf ("You have already taken Physics quiz. Choose another option.\n");
                Sleep(1000);
            }
            break;

    case 3: if (chemistry_quiz == 0)
            {   system ("cls");
                chemistry_correct = chemistry();
                chemistry_quiz = 1;
            }
            else
            {   printf ("You have already taken Chemistry quiz. Choose another option.\n");
                Sleep(1000);
            }
            break;
    case 4: if (history_quiz == 0)
            {   system ("cls");
                history_correct = history();
                history_quiz = 1;
            }
            else
            {   printf ("You have already taken History quiz. Choose another option.\n");
                Sleep(1000);
            }
            break;
    case 5: if (biology_quiz == 0)
            {   system ("cls");
                biology_correct = biology();
                biology_quiz = 1;
            }
            else
            {   printf ("You have already taken Biology quiz. Choose another option.\n");
                Sleep(1000);
            }
            break;
    default: printf ("Please enter a valid option.\n");
             Sleep (1000);
             break;

}
system ("cls");
}
int num_of_mathquestions = nof_maths();
int num_of_physicsquestions = nof_physics();
int  num_of_chemistryquestions =  nof_chemistry();
int num_of_historyquestions = nof_history();
int num_of_biologyquestions = nof_biology();



float maths_percentage;
maths_percentage = (float)maths_correct/(float)num_of_mathquestions; // Calculate percentage for maths.

float physics_percentage;
physics_percentage = (float)physics_correct/(float)num_of_physicsquestions; //Calculate percentage for physics.

float chemistry_percentage;
chemistry_percentage = (float)chemistry_correct/(float)num_of_chemistryquestions; //Calculate percentage for chemistry.

float history_percentage;
history_percentage = (float)history_correct/(float)num_of_historyquestions; //Calculate percentage for chemistry.

float biology_percentage;
biology_percentage = (float)biology_correct/(float)num_of_biologyquestions; //Calculate percentage for chemistry.

float total_percentage = (maths_percentage+physics_percentage+chemistry_percentage+history_percentage+biology_percentage)/5; // Calculate total percentage.

/*The result is printed when loop is exited.
In the result, grade function is used to calculate the grade for a given percentage.*/

/* Used printf statement to display the result to user on console. */

system ("cls");
printf ("           +-+-+-+-+-+-+\n");
printf ("           |R|E|S|U|L|T|\n");
printf ("           +-+-+-+-+-+-+\n\n");
printf ("Student Name:\t %s\n", full_name);
printf ("Roll No:\t %s\n\n", roll_no);

printf ("Subject\t\tScore\t\tGrade\n\n");
printf ("Math\t\t%d/%d\t\t%c\n",maths_correct, num_of_mathquestions,grade(maths_percentage));
printf ("Physics\t\t%d/%d\t\t%c\n",physics_correct, num_of_physicsquestions,grade(physics_percentage));
printf ("Chemistry\t%d/%d\t\t%c\n",chemistry_correct, num_of_chemistryquestions,grade (chemistry_percentage));
printf ("History\t\t%d/%d\t\t%c\n",history_correct, num_of_historyquestions,grade (history_percentage));
printf ("Biology\t\t%d/%d\t\t%c\n",biology_correct, num_of_biologyquestions,grade (biology_percentage));
printf ("****************************************\n");
printf ("Total\t\t%d/%d\t\t%c\n",maths_correct+physics_correct+chemistry_correct+history_correct+biology_correct, num_of_chemistryquestions+num_of_mathquestions+num_of_physicsquestions+num_of_historyquestions+num_of_biologyquestions, grade(total_percentage));
printf ("****************************************\n");

/*Also use fprintf to save the results into a file.*/
FILE *fp; // Declare file pointer fp.
fp = fopen ("result.txt", "w+"); // Save the results in result.txt file.

fprintf (fp, "Student Name:\t %s\n", full_name);
fprintf (fp, "Roll No:\t %s\n\n", roll_no);

fprintf (fp, "Subject\t\tScore\t\tGrade\n\n");
fprintf (fp,"Math\t\t%d/%d\t\t%c\n",maths_correct, num_of_mathquestions,grade(maths_percentage));
fprintf (fp, "Physics\t\t%d/%d\t\t%c\n",physics_correct, num_of_physicsquestions,grade(physics_percentage));
fprintf (fp, "Chemistry\t%d/%d\t\t%c\n",chemistry_correct, num_of_chemistryquestions,grade (chemistry_percentage));
fprintf (fp, "History\t\t%d/%d\t\t%c\n",history_correct, num_of_historyquestions,grade (history_percentage));
fprintf (fp, "Biology\t\t%d/%d\t\t%c\n",biology_correct, num_of_biologyquestions,grade (biology_percentage));
fprintf (fp, "****************************************\n");
fprintf (fp, "Total\t\t%d/%d\t\t%c\n",maths_correct+physics_correct+chemistry_correct+history_correct+biology_correct, num_of_chemistryquestions+num_of_mathquestions+num_of_physicsquestions+num_of_historyquestions+num_of_biologyquestions, grade(total_percentage));
fprintf (fp, "****************************************\n");

fclose (fp);

system ("pause");

return 0;
}


questions.h
/*These functions calculate the number of questions which are stored in
a text file. There are 7 lines for each question. Once the pointer goes through
7 lines, it constitutes a question*/


// To calculate number of math questions.
int nof_maths (void)
{
int i=0 ,j=0;
char maths_question[20][7][200]; // Declare an array to store math questions and options also correct answer.

FILE * fp;
fp = fopen("maths.txt", "r");

while (!feof(fp))
{
    for (j=0; j<7; j++)
    {
        fgets (maths_question[i][j], 200, fp);
        strtok(maths_question[i][j], "\n");
    }
    i++;
}
fclose (fp);

return i;
}

// To calculate the number of physics questions.
int nof_physics (void)
{
int i=0 ,j=0;
char physics_question[20][7][200]; // Declare an array to store math questions and options also correct answer.

FILE * fp;
fp = fopen("physics.txt", "r");

while (!feof(fp))
{
    for (j=0; j<7; j++)
    {
        fgets (physics_question[i][j], 200, fp);
        strtok(physics_question[i][j], "\n");
    }
    i++;
}
fclose (fp);

return i;
}

//To calculate the number of chemistry questions.
int nof_chemistry (void)
{
int i=0 ,j=0;
char chemistry_question[20][7][200]; // Declare an array to store math questions and options also correct answer.

FILE * fp;
fp = fopen("chemistry.txt", "r");

while (!feof(fp))
{
    for (j=0; j<7; j++)
    {
        fgets (chemistry_question[i][j], 200, fp);
        strtok(chemistry_question[i][j], "\n");
    }
    i++;
}
fclose (fp);

return i;
}


count_questions.h
/*These functions calculate the number of questions which are stored in
a text file. There are 7 lines for each question. Once the pointer goes through
7 lines, it constitutes a question*/


// To calculate number of math questions.
int nof_maths (void)
{
int i=0 ,j=0;
char maths_question[20][7][200]; // Declare an array to store math questions and options also correct answer.

FILE * fp;
fp = fopen("maths.txt", "r");

while (!feof(fp)) // File end of file function is used so loop runs until the pointer reaches the end of file.
{
    for (j=0; j<7; j++)
    {
        fgets (maths_question[i][j], 200, fp);
        strtok(maths_question[i][j], "\n"); //strtok function is used to eliminate the new line '\n' from the end of string.
    }
    i++;
}
fclose (fp);

return i;
}

// To calculate the number of physics questions.
int nof_physics (void)
{
int i=0 ,j=0;
char physics_question[20][7][200]; // Declare an array to store physics questions and options also correct answer.

FILE * fp;
fp = fopen("physics.txt", "r");

while (!feof(fp)) // File end of file function is used so loop runs until the pointer reaches the end of file.
{
    for (j=0; j<7; j++)
    {
        fgets (physics_question[i][j], 200, fp);
        strtok(physics_question[i][j], "\n"); // Strtok function is used to remove '\n' from the end of string.
    }
    i++;
}
fclose (fp);

return i;
}

//To calculate the number of chemistry questions.
int nof_chemistry (void)
{
int i=0 ,j=0;
char chemistry_question[20][7][200]; // Declare an array to store math questions and options also correct answer.

FILE * fp;
fp = fopen("chemistry.txt", "r");

while (!feof(fp)) // File end of file function is used so loop runs until the pointer reaches the end of file.
{
    for (j=0; j<7; j++)
    {
        fgets (chemistry_question[i][j], 200, fp);
        strtok(chemistry_question[i][j], "\n"); //Strtok function is used to remove the '\n' character from end of string.
    }
    i++;
}
fclose (fp);

return i;
}

int nof_history (void)
{
int i=0 ,j=0;
char history_question[20][7][200]; // Declare an array to store math questions and options also correct answer.

FILE * fp;
fp = fopen("history.txt", "r");

while (!feof(fp)) // File end of file function is used so loop runs until the pointer reaches the end of file.
{
    for (j=0; j<7; j++)
    {
        fgets (history_question[i][j], 200, fp);
        strtok(history_question[i][j], "\n"); //Strtok function is used to remove the '\n' character from end of string.
    }
    i++;
}
fclose (fp);

return i;
}

int nof_biology (void)
{
int i=0 ,j=0;
char biology_question[20][7][200]; // Declare an array to store math questions and options also correct answer.

FILE * fp;
fp = fopen("biology.txt", "r");

while (!feof(fp)) // File end of file function is used so loop runs until the pointer reaches the end of file.
{
    for (j=0; j<7; j++)
    {
        fgets (biology_question[i][j], 200, fp);
        strtok(biology_question[i][j], "\n"); //Strtok function is used to remove the '\n' character from end of string.
    }
    i++;
}
fclose (fp);

return i;
}

functions.h
int maths (void)
{

int i=0,j=0;
char maths_question[20][7][200]; // Declare an array to store math questions and options also correct answer.
char maths_answer [20][2];

FILE * fp;
fp = fopen("maths.txt", "r");

if (fp == NULL)
{
    printf ("Error: Can not open maths.txt\n");
    Sleep (2500);
}
else
{
    while (!feof(fp)) // file end of file function returns 1 when file pointer reaches the end of file.
{
    for (j=0; j<7; j++)
    {
        fgets (maths_question[i][j], 200, fp);
        strtok(maths_question[i][j], "\n");
    }
    i++;
}
fclose(fp);
}
int no_of_mathquestions = i;
int maths_correct;
maths_correct = 0;

int* random_number = malloc (no_of_mathquestions *sizeof (int));

for (int i = 0; i < no_of_mathquestions; i++) //Populate the array with question numbers.
{
    random_number[i] = i;
}

for (int i = 0; i < no_of_mathquestions; i++) // Shuffle the array using rand function.
{
    int temp = random_number[i];
    int randomIndex = rand() % (no_of_mathquestions);

    random_number[i] = random_number[randomIndex];
    random_number[randomIndex] = temp;
}

for(i = 0; i<no_of_mathquestions; i++)
{
    int question_number = random_number [i];
    printf ("Question %d : %s\n", i+1, maths_question[question_number][0]);
    printf ("A. %s\n", maths_question[question_number][1]);
    printf ("B. %s\n", maths_question[question_number][2]);
    printf ("C. %s\n", maths_question[question_number][3]);
    printf ("D. %s\n", maths_question[question_number][4]);

    printf ("\nEnter your answer:  ");
    scanf ("%s", maths_answer[question_number]);

    //Loop to ensure that user enters a valid option. The loop is executed until user enters a,b,c or d.
    while (strcmp(_strupr(maths_answer[question_number]),"A")&&strcmp(_strupr(maths_answer[question_number]),"B")&&strcmp(_strupr(maths_answer[question_number]),"C")&&strcmp(_strupr(maths_answer[question_number]),"D"))
    {
        printf ("\nPlease Enter a valid option: ");
        scanf ("%s", maths_answer[question_number]);
    }

    // strcmp function is used to compare the value entered by user and the correct answer. It returns 0 when both strings are same.
    if (!strcmp(_strupr(maths_answer[question_number]),maths_question[question_number][5]))
    {
        printf ("\nCorrect Answer. Points +1\n");
        maths_correct++;
        Sleep (1000);
    }
    else
    {
        printf ("\nWrong answer. \n");
        Sleep (1000);
    }
    getchar();
    system ("cls");

}

return maths_correct;

}


int physics (void)
{

int i=0,j=0;
char physics_question[20][7][150]; // Declare an array to store physics questions and options also correct answer.
char physics_answer [20][2];
int physics_correct = 0;

FILE * fp;
fp = fopen("physics.txt", "r");

if (fp == NULL)
{
    printf ("Error: Can not open physics.txt");
    Sleep (2500);
}
else
{
    while (!feof(fp)) // file end of file pointer return 1 when file pointer is at the end of file.
{
    for (j=0; j<7; j++)
    {
        fgets (physics_question[i][j], 150, fp);
        strtok(physics_question[i][j], "\n");
    }
    i++;
}
fclose(fp);

int no_of_physicsquestions = i;
int* random_number = malloc(no_of_physicsquestions * sizeof(int));

for (int i = 0; i < no_of_physicsquestions; i++) //Populate the array with question numbers.
{
    random_number[i] = i;
}

for (int i = 0; i < no_of_physicsquestions; i++) // Shuffle the array using rand function.
{
    int temp = random_number[i];
    int randomIndex = rand() % (no_of_physicsquestions);

    random_number[i] = random_number[randomIndex];
    random_number[randomIndex] = temp;
}



for(i = 0; i<no_of_physicsquestions; i++)
{
    int question_number = random_number [i];
    printf ("Question %d : %s\n", i+1, physics_question[question_number][0]);
    printf ("A. %s\n", physics_question[question_number][1]);
    printf ("B. %s\n", physics_question[question_number][2]);
    printf ("C. %s\n", physics_question[question_number][3]);
    printf ("D. %s\n", physics_question[question_number][4]);

    printf ("\nEnter your answer:  ");
    scanf ("%s", physics_answer[question_number]);

    //Loop to ensure that user enters a valid option. The loop is executed until user enters a,b,c or d.
    while (strcmp(_strupr(physics_answer[question_number]),"A")&&strcmp(_strupr(physics_answer[question_number]),"B")&&strcmp(_strupr(physics_answer[question_number]),"C")&&strcmp(_strupr(physics_answer[question_number]),"D"))
    {
        printf ("\nPlease Enter a valid option: ");
        scanf ("%s", physics_answer[question_number]);
    }

    // strcmp function is used to compare the value entered by user and the correct answer. It returns 0 when both strings are same.
    if (!strcmp(_strupr(physics_answer[question_number]),physics_question[question_number][5]))
    {
        printf ("\nCorrect Answer. Points +1\n");
        physics_correct++;
        Sleep (1000);
    }
    else
    {
        printf ("\nWrong answer. \n");
        Sleep (1000);
    }
    getchar();
    system ("cls");
} // End of for loop.
} // End of if..else statement used for file pointer. Else statement runs when fp opens physics.txt file.

return physics_correct;

}

int chemistry (void)
{

int i=0,j=0;
char chemistry_question[20][7][200]; // Declare an array to store chemistry questions and options also correct answer.
char chemistry_answer [20][2];
int chemistry_correct = 0;

FILE * fp;
fp = fopen("chemistry.txt", "r"); // Open maths.txt

if (fp == NULL)
{
    printf ("Error : Can not open chemistry.txt");
    Sleep (2500);
}
else
{
while (!feof(fp)) // file end of file function returns 1 when file pointer reaches the end of file.
{
    for (j=0; j<7; j++)
    {
        fgets (chemistry_question[i][j], 200, fp); // scan chemistry.txt line by line and store the strings into questions array.
        strtok(chemistry_question[i][j], "\n"); // strtok function is used to remove the \n from the en of each string.
    }
    i++;
}
fclose(fp);
int no_of_chemistryquestions = i; // The number of chemistry questions is equal to the no of times the while loop runs.
int* random_number = malloc(no_of_chemistryquestions * sizeof(int));;

for (int i = 0; i < no_of_chemistryquestions; i++) //Populate the array with question numbers.
{
    random_number[i] = i;
}

for (int i = 0; i < no_of_chemistryquestions; i++) // Shuffle the array using rand function.
{
    int temp = random_number[i];
    int randomIndex = rand() % (no_of_chemistryquestions);

    random_number[i] = random_number[randomIndex];
    random_number[randomIndex] = temp;
}



for(i = 0; i<no_of_chemistryquestions; i++)
{
    int question_number = random_number [i];
    printf ("Question %d : %s\n", i+1, chemistry_question[question_number][0]);
    printf ("A. %s\n", chemistry_question[question_number][1]);
    printf ("B. %s\n", chemistry_question[question_number][2]);
    printf ("C. %s\n", chemistry_question[question_number][3]);
    printf ("D. %s\n", chemistry_question[question_number][4]);

    printf ("\nEnter your answer:  ");
    scanf ("%s", chemistry_answer[question_number]);

    //Loop to ensure that user enters a valid option. The loop is executed until user enters a,b,c or d.
    while (strcmp(_strupr(chemistry_answer[question_number]),"A")&&strcmp(_strupr(chemistry_answer[question_number]),"B")&&strcmp(_strupr(chemistry_answer[question_number]),"C")&&strcmp(_strupr(chemistry_answer[question_number]),"D"))
    {
        printf ("\nPlease Enter a valid option: ");
        scanf ("%s", chemistry_answer[question_number]);
    }

    // strcmp function is used to compare the value entered by user and the correct answer. It returns 0 when both strings are same.
    if (!strcmp(_strupr(chemistry_answer[question_number]),chemistry_question[question_number][5]))
    {
        printf ("\nCorrect Answer. Points +1\n");
        chemistry_correct++;
        Sleep (1000);
    }
    else
    {
        printf ("\nWrong answer. \n");
        Sleep (1000);
    }
    getchar();
    system ("cls");
} // End of for loop.
} // End of if..else statement used for file pointer. Else statement runs when fp opens physics.txt file.

return chemistry_correct;

}

int history (void)
{

int i=0,j=0;
char history_question[20][7][200]; // Declare an array to store history questions and options also correct answer.
char history_answer [20][2];

FILE * fp;
fp = fopen("history.txt", "r");

if (fp == NULL)
{
    printf ("Error: Can not open history.txt\n");
    Sleep (2500);
}
else
{
    while (!feof(fp)) // file end of file function returns 1 when file pointer reaches the end of file.
{
    for (j=0; j<7; j++)
    {
        fgets (history_question[i][j], 200, fp);
        strtok(history_question[i][j], "\n");
    }
    i++;
}
fclose(fp);
}
int no_of_historyquestions = i;
int history_correct;
history_correct = 0;

int* random_number = malloc (no_of_historyquestions *sizeof (int));

for (int i = 0; i < no_of_historyquestions; i++) //Populate the array with question numbers.
{
    random_number[i] = i;
}

for (int i = 0; i < no_of_historyquestions; i++) // Shuffle the array using rand function.
{
    int temp = random_number[i];
    int randomIndex = rand() % (no_of_historyquestions);

    random_number[i] = random_number[randomIndex];
    random_number[randomIndex] = temp;
}

for(i = 0; i<no_of_historyquestions; i++)
{
    int question_number = random_number [i];
    printf ("Question %d : %s\n", i+1, history_question[question_number][0]);
    printf ("A. %s\n", history_question[question_number][1]);
    printf ("B. %s\n", history_question[question_number][2]);
    printf ("C. %s\n", history_question[question_number][3]);
    printf ("D. %s\n", history_question[question_number][4]);

    printf ("\nEnter your answer:  ");
    scanf ("%s", history_answer[question_number]);

    //Loop to ensure that user enters a valid option. The loop is executed until user enters a,b,c or d.
    while (strcmp(_strupr(history_answer[question_number]),"A")&&strcmp(_strupr(history_answer[question_number]),"B")&&strcmp(_strupr(history_answer[question_number]),"C")&&strcmp(_strupr(history_answer[question_number]),"D"))
    {
        printf ("\nPlease Enter a valid option: ");
        scanf ("%s", history_answer[question_number]);
    }

    // strcmp function is used to compare the value entered by user and the correct answer. It returns 0 when both strings are same.
    if (!strcmp(_strupr(history_answer[question_number]),history_question[question_number][5]))
    {
        printf ("\nCorrect Answer. Points +1\n");
        history_correct++;
        Sleep (1000);
    }
    else
    {
        printf ("\nWrong answer. \n");
        Sleep (1000);
    }
    getchar();
    system ("cls");

}

return history_correct;

}

int biology (void)
{

int i=0,j=0;
char biology_question[20][7][200]; // Declare an array to store biology questions and options also correct answer.
char biology_answer [20][2];

FILE * fp;
fp = fopen("biology.txt", "r");

if (fp == NULL)
{
    printf ("Error: Can not open biology.txt\n");
    Sleep (2500);
}
else
{
    while (!feof(fp)) // file end of file function returns 1 when file pointer reaches the end of file.
{
    for (j=0; j<7; j++)
    {
        fgets (biology_question[i][j], 200, fp);
        strtok(biology_question[i][j], "\n");
    }
    i++;
}
fclose(fp);
}
int no_of_biologyquestions = i;
int biology_correct;
biology_correct = 0;

int* random_number = malloc (no_of_biologyquestions *sizeof (int));

for (int i = 0; i < no_of_biologyquestions; i++) //Populate the array with question numbers.
{
    random_number[i] = i;
}

for (int i = 0; i < no_of_biologyquestions; i++) // Shuffle the array using rand function.
{
    int temp = random_number[i];
    int randomIndex = rand() % (no_of_biologyquestions);

    random_number[i] = random_number[randomIndex];
    random_number[randomIndex] = temp;
}

for(i = 0; i<no_of_biologyquestions; i++)
{
    int question_number = random_number [i];
    printf ("Question %d : %s\n", i+1, biology_question[question_number][0]);
    printf ("A. %s\n", biology_question[question_number][1]);
    printf ("B. %s\n", biology_question[question_number][2]);
    printf ("C. %s\n", biology_question[question_number][3]);
    printf ("D. %s\n", biology_question[question_number][4]);

    printf ("\nEnter your answer:  ");
    scanf ("%s", biology_answer[question_number]);

    //Loop to ensure that user enters a valid option. The loop is executed until user enters a,b,c or d.
    while (strcmp(_strupr(biology_answer[question_number]),"A")&&strcmp(_strupr(biology_answer[question_number]),"B")&&strcmp(_strupr(biology_answer[question_number]),"C")&&strcmp(_strupr(biology_answer[question_number]),"D"))
    {
        printf ("\nPlease Enter a valid option: ");
        scanf ("%s", biology_answer[question_number]);
    }

    // strcmp function is used to compare the value entered by user and the correct answer. It returns 0 when both strings are same.
    if (!strcmp(_strupr(biology_answer[question_number]),biology_question[question_number][5]))
    {
        printf ("\nCorrect Answer. Points +1\n");
        biology_correct++;
        Sleep (1000);
    }
    else
    {
        printf ("\nWrong answer. \n");
        Sleep (1000);
    }
    getchar();
    system ("cls");

}

return biology_correct;

}

